mesonbuild/interpreter.py

```diff
@@ -874,7 +875,28 @@ def get_define_method(self, args, kwargs):
         extra_args = self.determine_args(kwargs)
         deps = self.determine_dependencies(kwargs)
         value = self.compiler.get_define(element, prefix, self.environment, extra_args, deps)
-        mlog.log('Checking for value of define "%s": %s' % (element, value))
+        mlog.log('Fetching value of define "%s": %s' % (element, value))
+        return value
+
+    def get_return_value_method(self, args, kwargs):
+        if len(args) != 1:
+            raise InterpreterException('get_return_value() requires exactly '
+                                       'one argument, the function name')
+        check_stringlist(args)
+        fname = args[0]
+        rtype = kwargs.get('return_type', 'string')
+        if rtype not in ('string', 'int'):
+            m = "Unknown value {!r} for option 'return_type', valid values " \
+                "are 'string' and 'int'"
+            raise InterpreterException(m.format(rtype))
+        prefix = kwargs.get('prefix', '')
+        if not isinstance(prefix, str):
+            raise InterpreterException('Prefix argument of get_return_value() must be a string')
+        extra_args = self.determine_args(kwargs)
+        deps = self.determine_dependencies(kwargs)
+        value = self.compiler.get_return_value(fname, rtype, prefix, self.environment,
+                                               extra_args, deps)
+        mlog.log('Fetching return value of {!r}(): {!r}'.format(fname, value))
```

Comment:
Yes, I just wanted to get some feedback about the whole thing first. This method is very useful internally, but I am not sure how useful it is as an external API.